homeassistant/components/powerwall/config_flow.py

```diff
@@ -32,19 +33,24 @@
 }
 
 
-def _login_and_fetch_site_info(
+async def _login_and_fetch_site_info(
     power_wall: Powerwall, password: str
-) -> tuple[SiteInfo, str]:
+) -> tuple[SiteInfoResponse, str]:
     """Login to the powerwall and fetch the base info."""
     if password is not None:
-        power_wall.login(password)
-    return power_wall.get_site_info(), power_wall.get_gateway_din()
+        await power_wall.login(password)
 
+    return cast(
```

Comment:
Please guard casts with if TYPE_CHECKING (especially ones that construct generics each time)

Reply:
It would be better if the library had a py.typed marker file

Reply:
I'm a python beginner.  😊  I'll look that up.   I had no idea why the linter was getting confused here.

Reply:
https://peps.python.org/pep-0561/

Reply:
> It would be better if the library had a py.typed marker file

Hmm, the library does have py.typed already.   When I try instructions on forcing it to be included in the [project.toml](https://github.com/jrester/tesla_powerwall/blob/master/pyproject.toml) then the HA linter still complains.   Perhaps this is because I'm using a local install of the package... who knows.

Reply:
After the new `tesla_powerwall` library was released the type checking now works.   Removed the casts